app/posts/[url]/page.tsx

```diff
@@ -0,0 +1,34 @@
+import { allPosts } from '@/.contentlayer/generated';
+
+interface PageProps {
+  params: {
+    url: string;
+  };
+}
+
+export async function generateStaticParams(): Promise<PageProps['params'][]> {
+  return allPosts.map(({ url }) => ({
+    url: url,
+  }));
+}
+
+export default function PostPage({ params }: PageProps) {
+  const post = allPosts.find(
+    ({ url }) => url.replaceAll('/posts/', '') === params.url,
```

Comment:
`replaceAll` is not going to work if we change pages structure. For example, changing the URL to `blog/posts/post-name` will lead to the result of `replaceAll('/posts/', '')` to be `blog/post-name`. To remove the need for maintenance in case of changes here, we can use [endsWith](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith)